mllib/src/main/scala/org/apache/spark/ml/clustering/BisectingKMeans.scala

```diff
@@ -125,8 +125,13 @@ class BisectingKMeansModel private[ml] (
   /**
    * Computes the sum of squared distances between the input points and their corresponding cluster
    * centers.
+   *
+   * @deprecated This method is deprecated and will be removed in 3.0.0. Use ClusteringEvaluator
+   *             instead. You can also get the cost on the training dataset in the summary.
    */
   @Since("2.0.0")
+  @deprecated("This method is deprecated and will be removed in 3.0.0. Use ClusteringEvaluator " +
```

Comment:
I'm wondering if this PR is a blocker for Spark 2.4. According to JIRA desciption (Improvement/Minor), we cannot remove this 3.0.0 because we didn't announce deprecation before 3.0.0.

cc @cloud-fan since he is a release manager.

Reply:
It looks reasonable to me to deprecate it in 2.4 so that we can remove it in 3.0, if this is the last one. Then we can have a consistent ML API in 3.0 after removing these deprecated APIs.

Reply:
Thank you for the decision, @cloud-fan !

Reply:
yes this is the last one.

> Then we can have a consistent ML API in 3.0 after removing these deprecated APIs.

Yes, that's my goal in targeting this for 2.4.

Thanks.